pcdsdevices/sample_delivery.py

```diff
@@ -0,0 +1,440 @@
+"""
+Module for the Sample Delivery System and related devices.
+"""
+from ophyd.device import Component as Cpt
+from ophyd.device import Device
+from ophyd.signal import EpicsSignal
+
+from .component import UnrelatedComponent as UCpt
+from .signal import PytmcSignal
+
+
+class ViciValve(Device):
+    """
+    A Vici Valve as used in the SDS Selector.
+
+    Parameters
+    ----------
+    prefix : str
+        The base PV for the Selector.
+
+    name : str
+        A name for the device.
+    """
+
+    req_pos = Cpt(PytmcSignal, ':ReqPos', io='io', kind='normal')
+    curr_pos = Cpt(PytmcSignal, ':CurrentPos', io='i', kind='normal')
+
+
+class Selector(Device):
+    """
+    A Selector for the sample delivery system.
+
+    Parameters
+    ----------
+    prefix : str
+        The base PV for the Selector.
+
+    name : str
+        A name for the device.
+
+    names_button_prefix : str
+
+    couple_button_prefix : str
+
+    names1_prefix : str
+
+    names2_prefix : str
+    """
+
+    status = Cpt(PytmcSignal, ':IO:SyncUnitOK', io='i', kind='normal')
+
+    sampleFM_flow = Cpt(PytmcSignal, ':SampleFM:Flow', io='i', kind='normal')
+    sampleFM_state = Cpt(PytmcSignal, ':SampleFM:State', io='i',
+                         kind='normal')
+    sampleFM_reset = Cpt(PytmcSignal, ':SampleFM:Reset', io='o', kind='normal')
+    sampleFM_mode = Cpt(PytmcSignal, ':SampleFM:Mode', io='o', kind='normal')
+    sampleFM_mode_rb = Cpt(PytmcSignal, ':SampleFM:ModeRb', io='i',
+                           kind='normal')
+
+    sheathFM_flow = Cpt(PytmcSignal, ':SheathFM:Flow', io='i', kind='normal')
+    sheathFM_state = Cpt(PytmcSignal, ':SheathFM:State', io='i',
+                         kind='normal')
+    sheathFM_reset = Cpt(PytmcSignal, ':SheathFM:Reset', io='o', kind='normal')
+    sheathFM_mode = Cpt(PytmcSignal, ':SheathFM:Mode', io='o', kind='normal')
+    sheathFM_mode_rb = Cpt(PytmcSignal, ':SheathFM:ModeRb', io='i',
+                           kind='normal')
+
+    massFM_unit = Cpt(PytmcSignal, ':MassFM:Unit', io='i', kind='normal')
+    massFM_flow = Cpt(PytmcSignal, ':MassFM:Flow', io='i', kind='normal')
+
+    # TODO: Add CXI:SDS:SEL1:SYNC_RES_REQ and other aux records
+
+    shaker1 = Cpt(PytmcSignal, ':Shaker1:Ctrl', io='o', kind='normal')
+    shaker2 = Cpt(PytmcSignal, ':Shaker1:Ctrl', io='o', kind='normal')
+    shaker3 = Cpt(PytmcSignal, ':Shaker1:Ctrl', io='o', kind='normal')
+    shaker4 = Cpt(PytmcSignal, ':Shaker1:Ctrl', io='o', kind='normal')
+
+    valve1 = Cpt(ViciValve, ':Valve:01', name='ViciValve1')
+    valve2 = Cpt(ViciValve, ':Valve:02', name='ViciValve2')
```

Comment:
This is nitpicky but I think we should be consistent about how enumerated subcomponent PVs.  The shakers go according to `:Shaker1:` etc while these are `:Valve:01:` etc.  Not to be fixed here of course -- the pytmc pragmas are written this way in `ST_Selector`

Reply:
True, the valves have to be in this format as they are created as an array, but the shakers can be changed to match this format.